xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp

```diff
@@ -2732,6 +2703,35 @@ void CLinuxRendererGL::DeleteCLUT()
   }
 }
 
+void CLinuxRendererGL::CheckVideoParameters(int index)
+{
+  CPictureBuffer &buf = m_buffers[index];
+  int method = m_videoSettings.m_ToneMapMethod;
+
+  AVColorPrimaries srcPrim = GetSrcPrimaries(buf.m_srcPrimaries, buf.image.width, buf.image.height);
+  if (srcPrim != m_srcPrimaries)
+  {
+    m_srcPrimaries = srcPrim;
+    m_reloadShaders = true;
+  }
+
+  bool toneMap = false;
```

Comment:
Some HDR PQ videos has not metadata and they are not going to activate tone mapping. It is better to check that the stream transfer is `AVCOL_TRC_SMPTE2084` instead:

```
bool toneMap = (buf.color_transfer == AVCOL_TRC_SMPTE2084 && buf.primaries == AVCOL_PRI_BT2020 &&
    method != VS_TONEMAPMETHOD_OFF);
```